corehq/apps/api/resources/v0_5.py

```diff
@@ -326,6 +331,40 @@ def _update(cls, bundle, user_change_logger=None):
 
         return errors
 
+    def prepend_urls(self):
+        return [
+            url(r"^(?P<resource_name>%s)/(?P<pk>\w[\w/-]*)/activate/$" % self._meta.resource_name,
+                self.wrap_view('activate_user'), name="api_activate_user"),
+            url(r"^(?P<resource_name>%s)/(?P<pk>\w[\w/-]*)/deactivate/$" % self._meta.resource_name,
+                self.wrap_view('deactivate_user'), name="api_deactivate_user"),
+        ]
+
+    @location_safe
+    def activate_user(self, request, **kwargs):
+        return self._modify_user_status(request, **kwargs, active=True)
+
+    @location_safe
+    def deactivate_user(self, request, **kwargs):
+        return self._modify_user_status(request, **kwargs, active=False)
+
+    def _modify_user_status(self, request, active, **kwargs):
+        self.is_authenticated(request)
+
+        try:
+            user = CommCareUser.get(kwargs['pk'])
+            assert user.domain == kwargs['domain']
```

Comment:
This should rather be done using existing methods (and outside the 'try/catch'):
```
user = CommCareUser.get_by_user_id(kwargs['pk'], kwargs["domain"])
if not user:
    raise NotFound()
```